site/_data/lib/find.js

```diff
@@ -1,51 +1,42 @@
 const path = require('path');
 
-const cacheKey = Symbol('find-cache');
-const alreadyNotifiedKey = Symbol('already-notified');
+const urlCacheKey = Symbol('find-cache-url');
+const pathCacheKey = Symbol('find-cache-path');
+
+const defaultLocale = 'en';
 
 /**
- * Builds and uses a cache to ensure fast lookup of Eleventy pages.
+ * Builds and uses a cache to ensure fast lookup of Eleventy pages by
+ * a specified key.
  *
  * @param {EleventyCollectionItem[]} collection
- * @param {string} urlToFind
+ * @param {unique symbol} cacheKey
+ * @param {string} key
+ * @param {string} needle
  * @return {EleventyCollectionItem|undefined}
  */
-const internalFind = (collection, urlToFind) => {
-  /** @type {{[url: string]: EleventyCollectionItem}} */
+const internalFind = (collection, cacheKey, key, needle) => {
   let cache = collection[cacheKey];
   if (cache === undefined) {
     cache = {};
     for (const entry of collection) {
-      if (entry.url) {
-        cache[entry.url] = entry;
+      if (entry[key]) {
+        cache[entry[key]] = entry;
       }
     }
     collection[cacheKey] = cache;
   }
 
-  if (urlToFind in cache) {
-    return cache[urlToFind];
+  if (needle in cache) {
+    return cache[needle];
   }
 
   // Otherwise, be slow, since we think this is probably pretty rare.
-  const result = collection.find(item => item.url === urlToFind);
+  const result = collection.find(item => item[key] === needle);
```

Comment:
item[key] can be an object. Would comparing objects (by ref) work here?

Reply:
See comment above, removed.